app/controllers/links_controller.rb

```diff
@@ -38,6 +40,22 @@ def documentation_url(section = "", options = {})
     root_url + (section || '')
   end
 
+  # For new documentation at docs.theforeman.org
+  def docs_url(guide, chapter, flavor: nil)
+    version = SETTINGS[:version].short
+    # TODO: setting?
+    flavor ||= if plugin_present?('katello')
+                 'katello'
+               elsif File.exist?('/etc/debian_version')
```

Comment:
Small nitpick: can we cache the file access? I don't feel comfortable with accessing the filesystem each documentation click.

Reply:
I had thought about that and in https://github.com/theforeman/foreman/pull/8613 we talked about exactly that. I was thinking of making it a `SETTING` (see the TODO just above it). Any thoughts on that?

Reply:
`SETTING` is a good start. It depends if we have more behaviors that depend on the fact that we run on debian.
If there are, I would suggest a `SETTING` and a `Foreman.on_debian?` method. If this is the only case, I suppose we can use `SETTING[:docs_flavor]` and set it either to `deb` or `el`

Reply:
The reason I'd avoid `docs_flavor` as a setting is that in [foreman-documentation](https://github.com/theforeman/foreman-documentation) we have 3 flavors: `deb`, `el` and `katello`. I don't think we can detect katello is present when we determine `SETTING` so the value would be a lie. Perhaps `docs_os_flavor` could work.

Reply:
Sounds good. Katello would be detected in the code, but the os flavor would come from `SETTING`.
Another option to handle Katello is to create a method that would be overridden by Katello. 
Something like 
``` ruby
def flavor
  SETTING[:base_os_flavor]
end
```
and then in Katello
```
module DocsOverrides
  def flavor
    'katello'
  end
end

LinksController.prepend DocsOverrides
```
This would clean up the core from plugin dependency.

Reply:
I can see advantages of both. Current version is easier to understand completely without having to dig into plugins. Yours could be slightly faster for non-Katello, though I think that difference is tiny and the doc controller isn't that time sensitive so I'm leaning to taking the easier to understand benefit here.

Reply:
ACK, let's keep Katello in the code, but to make it easier to remove it later, let's refactor the `flavor` to a separate method.